### PR TITLE
Revert start /wait for msiexec and change nulls to ? in UTF-8 mode.

### PR DESCRIPTION
**Pull request by BillyONeal**
https://github.com/microsoft/vcpkg/pull/23367 was a failure because we still end up trying to download portable git; however as some combination of changes in attempt to debug we drop all msiexec output on the floor now.

This reverts change to add start /wait, replaces nulls with ?s.

### CODE REVIEW

**Comment by Neumann-A** [comment]
as far as I can tell it worked before #394 was merge right? So why not revert that change?

**Comment by BillyONeal** [comment]
> as far as I can tell it worked before #394 was merge right? So why not revert that change?

I don't know which of the 2 changes were the cause of the problem and I think this replace-nulls-with-? behavior is better overall.

**Comment by Neumann-A** [comment]
It definetly is #394 since @ras0219-msft used the utf16 changes in the boost PR and the output worked.

**Comment by Neumann-A** [comment]
Here is working output from the CI debug PR: https://dev.azure.com/vcpkg/public/_build/results?buildId=68216&view=logs&jobId=f5997e23-771a-528a-d03c-549a3c7aa976&j=f5997e23-771a-528a-d03c-549a3c7aa976&t=edd0f0a3-c2df-55e3-8a4f-65fe34ccebfa
which is using:
https://github.com/ras0219-msft/vcpkg-tool/commits/dev/roschuma/msiexec
which basically is fcbde269d1107280b7e70e667788cc333ac9f537 (+2 commits from the UTF16 changes)

**Comment by ras0219-msft** [comment]
The `start /wait` was the change that broke it for me. Additionally, I have not observed issues with not waiting long enough via just `cmd /c`.

**Comment by dg0yt** [comment]
> The start /wait was the change that broke it for me. Additionally, I have not observed issues with not waiting long enough via just cmd /c.

I'm sorry for the unintented side effect. The usage of `misexec` in vcpkg is limited, but my history with this tool is > 10 years. There were lot of strange `msi` packages and error contexts in that time.

**Comment by BillyONeal** [comment]
I simplified this a lot based on the feedback above; it preserves the ability to interpret as UTF-16 when we know that's the case, as we do for msiexec, but also keeps the null to ? replacement.

**Review by vicroms — APPROVED** [review]

**Pull request merged**
